src/content/post/2025-03-23-kuzu-mcp-server.md

```diff
@@ -0,0 +1,273 @@
+We'll start by defining a schema that captures relationships between players, the
+countries they represent, and the World Cups their countries have won. Open a Kuzu
+CLI and execute the following DDL commands:
+
+```sql
+CREATE NODE TABLE Player(name STRING PRIMARY KEY, goalsScored INT32);
+CREATE NODE TABLE Country(name STRING PRIMARY KEY, worldCupsWon INT32);
+CREATE NODE TABLE WorldCup(year INT32 PRIMARY KEY, host STRING);
+CREATE REL TABLE PLAYS_FOR(FROM Player TO Country);
+CREATE REL TABLE WON(FROM Country TO WorldCup);
+```
+
+Next, insert the data into the database using Cypher:
+```cypher
+// Create Country nodes
+CREATE (argentina:Country {name: "Argentina", worldCupsWon: 3})
+CREATE (france:Country {name: "France", worldCupsWon: 2})
+
+// Create Player nodes
+CREATE (messi:Player {name: "Lionel Messi", goalsScored: 112})
+CREATE (dimaria:Player {name: "Angel Di Maria", goalsScored: 31})
+CREATE (mbappe:Player {name: "Kylian Mbappe", goalsScored: 48})
+CREATE (griezmann:Player {name: "Antoine Griezmann", goalsScored: 44})
+
+// Create WorldCup nodes
+CREATE (wc2022:WorldCup {year: 2022, host: "Qatar"})
+CREATE (wc2018:WorldCup {year: 2018, host: "Russia"})
+
+// Create PLAYS_FOR relationships
+CREATE (messi)-[:PLAYS_FOR]->(argentina)
+CREATE (dimaria)-[:PLAYS_FOR]->(argentina)
+CREATE (mbappe)-[:PLAYS_FOR]->(france)
+CREATE (griezmann)-[:PLAYS_FOR]->(france)
+
+// Create WON relationships
+CREATE (argentina)-[:WON]->(wc2022)
+CREATE (france)-[:WON]->(wc2018)
+```
+
+The database created using the above commands is stored in a directory named `football_db` on the local disk.
+
+## MCP server
+
+The official MCP [specification](https://spec.modelcontextprotocol.io/specification/2024-11-05/architecture/)
+released by Anthropic describes the protocol’s design principles in detail. The key qualities of an MCP server include:
+
+- **Simple and easy to build**: It should focus on specific, well-defined capabilities.
+- **Highly composable**: It should isolate capabilities so that multiple servers can be combined seamlessly.
+- **Maintain isolation**: Only the necessary contextual information should be exposed to the server.
+- **Progressively add features**: Additional capabilities can be added without breaking existing clients.
+
+Kuzu's MCP server is implemented in Node.js using the above principles, and exposes the following functions:
+| Function | Description |
+|----------|-------------|
+| `getSchema` | Returns the schema of the database |
+| `query` | Executes a Cypher query on the database |
+
+When you initialize a connection to the Kuzu MCP server, you open a read-write connection to the database,
+using which the client can send queries to the database via the server. The LLM that governs the MCP client
+decides which operations to perform based on the exposed functions in the MCP server.
+
+Note: Currently, Kuzu only supports MCP connections to local servers and databases. Remote databases are not yet supported.
```

Comment:
@mewim, do you think this sentence is correctly worded? I saw this in the SQLite blog post and it seemed relevant to us too.